local_planner/src/nodes/planner_functions.cpp

```diff
@@ -385,25 +390,28 @@ void findFreeDirections(
       }
 
       if (free) {
-        p.x = elevationIndexToAngle(e, resolution_alpha);
-        p.y = azimuthIndexToAngle(z, resolution_alpha);
-        p.z = 0.0;
+        p_pol = HistogramIndexToPolar(e, z, resolution_alpha, 0.0);
+        p.x = p_pol.e;
+        p.y = p_pol.z;
+        p.z = p_pol.r;
         path_candidates.cells.push_back(p);
         double cost =
-            costFunction((int)p.x, (int)p.y, path_waypoints, goal, position,
+            costFunction(p_pol.e, p_pol.z, path_waypoints, goal, position,
                          position_old, goal_cost_param, smooth_cost_param,
                          height_change_cost_param_adapted,
                          height_change_cost_param, only_yawed);
         cost_path_candidates.push_back(cost);
       } else if (!free && histogram.get_bin(e, z) != 0) {
-        p.x = elevationIndexToAngle(e, resolution_alpha);
-        p.y = azimuthIndexToAngle(z, resolution_alpha);
-        p.z = 0.0;
+        p_pol = HistogramIndexToPolar(e, z, resolution_alpha, 0.0);
+        p.x = p_pol.e;
+        p.y = p_pol.z;
+        p.z = p_pol.r;
         path_rejected.cells.push_back(p);
       } else {
-        p.x = elevationIndexToAngle(e, resolution_alpha);
-        p.y = azimuthIndexToAngle(z, resolution_alpha);
-        p.z = 0.0;
+        p_pol = HistogramIndexToPolar(e, z, resolution_alpha, 0.0);
+        p.x = p_pol.e;
+        p.y = p_pol.z;
+        p.z = p_pol.r;
         path_blocked.cells.push_back(p);
```

Comment:
If we're If we are sticking polar points into the `path_blocked.cells`, maybe it should be a vector of `PolarPoint`s?